src/styles/utilities.scss

```diff
@@ -14,22 +14,42 @@ $base: 16px !default;
   @return #{$rem}rem;
 }
 
-@mixin button-focus-ring {
-  outline: none;
-  box-shadow: 0 0 0 1px var(--bg-neutral-lightest), 0 0 0 var(--border-width-2) var(--bg-focus-ring);
-}
-
 @mixin input-focus-ring {
-  border-color: var(--color-border-input-selected);
-  outline: none;
-  box-shadow:
-    0 0 0 1px var(--bg-neutral-lightest),
-    0 0 0 var(--border-width-2) var(--bg-focus-ring),
-    inset 0 0 0 var(--border-width-0) var(--color-border-input-selected);
+  @media not all and (min-resolution: .001dpcm) { // Target only Safari browsers
+    @supports (-webkit-appearance:none) {         // Use `focus` instead of `focus-visible`
+      &:hover:focus,
+      &:focus {                                   // due to Safari's lack of support
+        border-color: var(--color-border-input-selected);
+        outline: none;
+        box-shadow:
+          0 0 0 1px var(--bg-body),
+          0 0 0 var(--border-width-2) var(--bg-focus-ring),
+          inset 0 0 0 var(--border-width-0) var(--color-border-input-selected);
+      }
+    }
+  }
+
+  &:focus-visible,
+  &:hover:focus-visible {
+    border-color: var(--color-border-input-selected);
+    outline: none;
+    box-shadow:
+      0 0 0 1px var(--bg-body),
+      0 0 0 var(--border-width-2) var(--bg-focus-ring),
+      inset 0 0 0 var(--border-width-0) var(--color-border-input-selected);
+  }
 }
 
-// Supports safari browsers
 @mixin focus-ring {
   outline: none;
-  box-shadow: 0 0 0 1px var(--bg-neutral-lightest), 0 0 0 var(--border-width-2) var(--bg-focus-ring);
+  box-shadow: 0 0 0 1px var(--bg-body), 0 0 0 var(--border-width-2) var(--bg-focus-ring);
+}
+
+@mixin focus-ring-visible {
```

Comment:
Hmm. So we will have:
- ` input-focus-ring`
- `focus-ring`
- `focus-ring-visible`


When to use `focus-ring-visible` or `focus-ring`? 👀

Reply:
We can remove the `focus-ring`, I'm gonna updated it here!

Reply:
Couldn't do the change for these components because they all have some special scenario:
- ProductCard (uses `focus-within`)
- SkuSelector (changes a `span` instead)
- QuantitySelector (changes the `[data-store-icon]` instead)